#Taller 2
Econometría ICS-294
I Semestre 2026

Profesor: Francisco Alfaro Medina

Ayudantes: Krischnna Cortez y Karen Rojas

Estudiante:Catalina Vera

In [4]:
# Instalar y cargar la librería wooldridge si no está instalada
if (!require(wooldridge)) install.packages("wooldridge")
library(wooldridge)

# Cargar la base de datos ceosal1
data("ceosal1", package = "wooldridge")
df_ceosal1 <- ceosal1

# Verificar si hay valores nulos
null_counts <- sum(is.na(df_ceosal1))

if (null_counts == 0) {
    print("La base de datos 'ceosal1' se ha cargado correctamente y no contiene valores nulos.")
} else {
    print(paste("La base de datos 'ceosal1' se ha cargado, pero contiene", null_counts, "valores nulos."))
    # Opcional: Si se desea eliminar las filas con nulos
    # df_ceosal1 <- na.omit(df_ceosal1)
    # print("Se han eliminado las filas con valores nulos.")
}

# Mostrar las primeras filas del DataFrame para verificar
print("Primeras 5 filas de la base de datos ceosal1:")
print(head(df_ceosal1))

Loading required package: wooldridge

Warning message in library(package, lib.loc = lib.loc, character.only = TRUE, logical.return = TRUE, :
“there is no package called ‘wooldridge’”
Installing package into ‘/usr/local/lib/R/site-library’
(as ‘lib’ is unspecified)



[1] "La base de datos 'ceosal1' se ha cargado correctamente y no contiene valores nulos."
[1] "Primeras 5 filas de la base de datos ceosal1:"
  salary pcsalary   sales  roe pcroe ros indus finance consprod utility
1   1095       20 27595.0 14.1 106.4 191     1       0        0       0
2   1001       32  9958.0 10.9 -30.6  13     1       0        0       0
3   1122        9  6125.9 23.5 -16.3  14     1       0        0       0
4    578       -9 16246.0  5.9 -25.7 -21     1       0        0       0
5   1368        7 21783.2 13.8  -3.0  56     1       0        0       0
6   1145        5  6021.4 20.0   1.0  55     1       0        0       0
   lsalary    lsales
1 6.998509 10.225389
2 6.908755  9.206132
3 7.022868  8.720281
4 6.359574  9.695602
5 7.221105  9.988894
6 7.043160  8.703075


In [6]:
# 1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico se emplea habitualmente esta base de datos?

# Para obtener una descripción de las variables y su estructura:
print("\nEstructura de la base de datos df_ceosal1:")
str(df_ceosal1)

print("\nResumen estadístico de las variables de df_ceosal1:")
summary(df_ceosal1)

[1] "\nEstructura de la base de datos df_ceosal1:"
'data.frame':	209 obs. of  12 variables:
 $ salary  : int  1095 1001 1122 578 1368 1145 1078 1094 1237 833 ...
 $ pcsalary: int  20 32 9 -9 7 5 10 7 16 5 ...
 $ sales   : num  27595 9958 6126 16246 21783 ...
 $ roe     : num  14.1 10.9 23.5 5.9 13.8 ...
 $ pcroe   : num  106.4 -30.6 -16.3 -25.7 -3 ...
 $ ros     : int  191 13 14 -21 56 55 62 44 37 37 ...
 $ indus   : int  1 1 1 1 1 1 1 1 1 1 ...
 $ finance : int  0 0 0 0 0 0 0 0 0 0 ...
 $ consprod: int  0 0 0 0 0 0 0 0 0 0 ...
 $ utility : int  0 0 0 0 0 0 0 0 0 0 ...
 $ lsalary : num  7 6.91 7.02 6.36 7.22 ...
 $ lsales  : num  10.23 9.21 8.72 9.7 9.99 ...
 - attr(*, "time.stamp")= chr "25 Jun 2011 23:03"
[1] "\nResumen estadístico de las variables de df_ceosal1:"


     salary         pcsalary          sales              roe       
 Min.   :  223   Min.   :-61.00   Min.   :  175.2   Min.   : 0.50  
 1st Qu.:  736   1st Qu.: -1.00   1st Qu.: 2210.3   1st Qu.:12.40  
 Median : 1039   Median :  9.00   Median : 3705.2   Median :15.50  
 Mean   : 1281   Mean   : 13.28   Mean   : 6923.8   Mean   :17.18  
 3rd Qu.: 1407   3rd Qu.: 20.00   3rd Qu.: 7177.0   3rd Qu.:20.00  
 Max.   :14822   Max.   :212.00   Max.   :97649.9   Max.   :56.30  
     pcroe            ros            indus           finance      
 Min.   :-98.9   Min.   :-58.0   Min.   :0.0000   Min.   :0.0000  
 1st Qu.:-21.2   1st Qu.: 21.0   1st Qu.:0.0000   1st Qu.:0.0000  
 Median : -3.0   Median : 52.0   Median :0.0000   Median :0.0000  
 Mean   : 10.8   Mean   : 61.8   Mean   :0.3206   Mean   :0.2201  
 3rd Qu.: 19.5   3rd Qu.: 81.0   3rd Qu.:1.0000   3rd Qu.:0.0000  
 Max.   :977.0   Max.   :418.0   Max.   :1.0000   Max.   :1.0000  
    consprod         utility          lsalary          

1. (10 puntos) Examine la base y elabore una breve descripción de sus variables, distinguiendo
entre dependientes, explicativas, continuas y dummies. ¿Para qué tipo de estudio económico
se emplea habitualmente esta base de datos?


1 set. seed (1)
2 data ( ceosal1 )
3 ceosal1 <- na. omit ( ceosal1 )
4 ? ceosal1

### Descripción de Variables de `ceosal1`

Esta base de datos se utiliza comúnmente en econometría para estudiar la **determinación del salario de los CEOs** y la relación entre el **rendimiento de la empresa y la compensación ejecutiva**.

**Variables Dependientes (ejemplos típicos):**
*   **salary**: Salario anual del CEO (en miles de dólares). *Continua.*
*   **lsalary**: Logaritmo natural del salario del CEO. *Continua.*

**Variables Explicativas:**
*   **pcsalary**: Cambio porcentual en el salario del CEO respecto al año anterior. *Continua.*
*   **sales**: Ventas anuales de la empresa (en millones de dólares). *Continua.*
*   **lsales**: Logaritmo natural de las ventas de la empresa. *Continua.*
*   **roe**: Retorno sobre el capital (Return on Equity) de la empresa (en porcentaje). *Continua.*
*   **pcroe**: Cambio porcentual en el ROE respecto al año anterior. *Continua.*
*   **ros**: Retorno sobre las acciones. *Continua.*
*   **indus**: Dummy = 1 si la empresa es del sector primario. *Dummy.*
*   **finance**: Dummy = 1 si la empresa es del sector financiero. *Dummy.*
*   **consprod**: Dummy = 1 si la empresa es de productos de consumo. *Dummy.*
*   **utility**: Dummy = 1 si la empresa es de servicios públicos. *Dummy.*

**Tipo de Estudio Económico Habitual:**
La base de datos `ceosal1` es ideal para **análisis de regresión de corte transversal** en **econometría corporativa** o **economía laboral**. Se usa para investigar factores que influyen en el salario de los altos ejecutivos, como el tamaño de la empresa (ventas), su rendimiento financiero (ROE), y las características de la industria, a menudo explorando la hipótesis del vínculo entre desempeño y compensación o problemas de agencia.

2. (5 puntos) Genere una variable indicadora (dummy) que tome el valor 1 cuando la variable
sales supere su promedio y 0 en caso contrario. Asigne a esta nueva variable un nombre que
sea fácilmente identificable.

In [8]:
df_ceosal1$sales_mayor_promedio <- ifelse(df_ceosal1$sales > mean(df_ceosal1$sales), 1, 0)

# Verificar las primeras filas con la nueva variable
print("Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio':")
print(head(df_ceosal1[, c("sales", "sales_mayor_promedio")]))

# Verificar la distribución de la nueva variable
print("Tabla de frecuencias para 'sales_mayor_promedio':")
print(table(df_ceosal1$sales_mayor_promedio))

[1] "Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio':"
    sales sales_mayor_promedio
1 27595.0                    1
2  9958.0                    1
3  6125.9                    0
4 16246.0                    1
5 21783.2                    1
6  6021.4                    0
[1] "Tabla de frecuencias para 'sales_mayor_promedio':"

  0   1 
155  54 


3. (5 puntos) Cree una variable de la multuplicación entre la variable dummy antes creada y la
variable indus. ¿Qué indica esta interacción?

In [12]:
df_ceosal1$sales_mayor_promedio_indus <- df_ceosal1$sales_mayor_promedio * df_ceosal1$indus

# Verificar las primeras filas con la nueva variable
print("Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio_indus':")
print(head(df_ceosal1[, c("sales_mayor_promedio", "indus", "sales_mayor_promedio_indus")]))

# Verificar la distribución de la nueva variable
print("Tabla de frecuencias para 'sales_mayor_promedio_indus':")
print(table(df_ceosal1$sales_mayor_promedio_indus))

[1] "Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio_indus':"
  sales_mayor_promedio indus sales_mayor_promedio_indus
1                    1     1                          1
2                    1     1                          1
3                    0     1                          0
4                    1     1                          1
5                    1     1                          1
6                    0     1                          0
[1] "Tabla de frecuencias para 'sales_mayor_promedio_indus':"

  0   1 
188  21 


La variable de interacción sales_mayor_promedio_indus indica si una empresa pertenece al sector 'indus' (donde indus es igual a 1) Y, al mismo tiempo, si sus ventas están por encima del promedio general (sales_mayor_promedio es igual a 1). Si ambas condiciones se cumplen, el valor de esta variable será 1; de lo contrario, será 0. En términos sencillos, nos ayuda a ver si el impacto de tener ventas altas en el salario de un CEO es diferente específicamente para las empresas en el sector 'indus', o si hay alguna particularidad en ese grupo.

 (5 puntos) Cree una variable de la multiplicación entre la variable continua roe y la variable
dummy antes creada. ¿Qué indica esta interacción?


In [16]:
df_ceosal1$sales_mayor_promedio_roe <- df_ceosal1$sales_mayor_promedio * df_ceosal1$roe

# Verificar las primeras filas con la nueva variable
print("Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio_roe':")
print(head(df_ceosal1[, c("sales_mayor_promedio", "roe", "sales_mayor_promedio_roe")]))

# Verificar un resumen estadístico de la nueva variable
print("\nResumen estadístico para 'sales_mayor_promedio_roe':")
print(summary(df_ceosal1$sales_mayor_promedio_roe))




[1] "Primeras 5 filas de df_ceosal1 con la nueva variable 'sales_mayor_promedio_roe':"
  sales_mayor_promedio  roe sales_mayor_promedio_roe
1                    1 14.1                     14.1
2                    1 10.9                     10.9
3                    0 23.5                      0.0
4                    1  5.9                      5.9
5                    1 13.8                     13.8
6                    0 20.0                      0.0
[1] "\nResumen estadístico para 'sales_mayor_promedio_roe':"
   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
  0.000   0.000   0.000   4.251   3.500  43.700 



Esta variable de interacción toma el valor de 'roe' cuando la empresa tiene ventas por encima del promedio ('sales_mayor_promedio' = 1). Si la empresa tiene ventas por debajo o igual al promedio ('sales_mayor_promedio' = 0), la variable de interacción será 0, por ende nos permite analizar el efecto del retorno sobre el capital ('roe') específicamente para aquellas empresas que también tienen un alto volumen de ventas (por encima del promedio). Es útil para ver si el impacto de 'roe' en una variable dependiente (como el salario del CEO) difiere para empresas con ventas altas versus bajas.


5. (5 puntos) Cree una variable que indique si el compañia corresponde a industria (indus) ó si
finance. Asigne a esta nueva variable un nombre que sea fácilmente identificable.



In [17]:
df_ceosal1$indus_o_finance <- df_ceosal1$indus + df_ceosal1$finance

# Verificar las primeras filas con la nueva variable
print("Primeras 5 filas de df_ceosal1 con la nueva variable 'indus_o_finance':")
print(head(df_ceosal1[, c("indus", "finance", "indus_o_finance")]))

# Verificar la distribución de la nueva variable
print("\nTabla de frecuencias para 'indus_o_finance':")
print(table(df_ceosal1$indus_o_finance))

# Opcional: Para asegurar que sea estrictamente 0 o 1, aunque con dummies 0/1 mutuamente excluyentes, la suma ya lo asegura.
# df_ceosal1$indus_o_finance[df_ceosal1$indus_o_finance > 1] <- 1
# print("\nTabla de frecuencias ajustada para 'indus_o_finance' (si hubiera valores > 1):")
# print(table(df_ceosal1$indus_o_finance))

[1] "Primeras 5 filas de df_ceosal1 con la nueva variable 'indus_o_finance':"
  indus finance indus_o_finance
1     1       0               1
2     1       0               1
3     1       0               1
4     1       0               1
5     1       0               1
6     1       0               1
[1] "\nTabla de frecuencias para 'indus_o_finance':"

  0   1 
 96 113 


2 Modelos de Regresión Lineal (70 puntos)
1. (15 puntos) Genere un modelo de regresión lineal, sobre alguna de las variables dependientes,
con respecto a las variables sales, roe y ros. Interprete sus resultados y la significancia de los
regresores. ¿El modelo tiene significancia global al 5%?

In [20]:

model_salary <- lm(lsalary ~ sales + roe + ros, data = df_ceosal1)


print("\nResumen del Modelo de Regresión Lineal (lsalary ~ sales + roe + ros):")
summary_model <- summary(model_salary)
print(summary_model)

# Evaluar la significancia global al 5%
if (summary_model$fstatistic[3] < 0.05) {
    print("\nEl modelo es globalmente significativo al 5% (p-valor del F-statistic < 0.05).")
} else {
    print("\nEl modelo NO es globalmente significativo al 5% (p-valor del F-statistic >= 0.05).")
}


[1] "\nResumen del Modelo de Regresión Lineal (lsalary ~ sales + roe + ros):"

Call:
lm(formula = lsalary ~ sales + roe + ros, data = df_ceosal1)

Residuals:
    Min      1Q  Median      3Q     Max 
-1.3299 -0.2933 -0.0390  0.2540  2.6480 

Coefficients:
              Estimate Std. Error t value Pr(>|t|)    
(Intercept)  6.612e+00  8.886e-02  74.409  < 2e-16 ***
sales        1.496e-05  3.484e-06   4.295  2.7e-05 ***
roe          1.683e-02  4.481e-03   3.757 0.000224 ***
ros         -8.801e-04  5.642e-04  -1.560 0.120331    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5292 on 205 degrees of freedom
Multiple R-squared:  0.1397,	Adjusted R-squared:  0.1271 
F-statistic:  11.1 on 3 and 205 DF,  p-value: 8.832e-07

[1] "\nEl modelo NO es globalmente significativo al 5% (p-valor del F-statistic >= 0.05)."


el modelo de regresión lineal es globalmente significativo al 5%. Esto se confirma porque el p-valor asociado al estadístico F es 8.832e-07, un valor considerablemente menor que el nivel de significancia de 0.05.

Que el modelo sea globalmente significativo implica que al menos una de las variables explicativas incluidas en el modelo tiene un efecto estadísticamente significativo sobre la variable dependiente (lsalary).

2. (10 puntos) Al modelo de regresión lineal anterior, agregue las variables creadas en el inciso 1.2
y 1.4, interprete sus resultados. ¿Cómo afectan los regresores de estas variables en el salario?

In [22]:
model_salary_extended <- lm(lsalary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe, data = df_ceosal1)

print("\nResumen del Modelo de Regresión Lineal Extendido:")
summary_model_extended <- summary(model_salary_extended)
print(summary_model_extended)

# Evaluar la significancia global al 5%
if (summary_model_extended$fstatistic[3] < 0.05) {
    print("\nEl modelo extendido es globalmente significativo al 5% (p-valor del F-statistic < 0.05).")
} else {
    print("\nEl modelo extendido NO es globalmente significativo al 5% (p-valor del F-statistic >= 0.05).")
}


[1] "\nResumen del Modelo de Regresión Lineal Extendido:"

Call:
lm(formula = lsalary ~ sales + roe + ros + sales_mayor_promedio + 
    sales_mayor_promedio_roe, data = df_ceosal1)

Residuals:
     Min       1Q   Median       3Q      Max 
-1.32626 -0.27365 -0.04448  0.24771  2.70138 

Coefficients:
                           Estimate Std. Error t value Pr(>|t|)    
(Intercept)               6.551e+00  9.773e-02  67.024  < 2e-16 ***
sales                     5.783e-06  4.241e-06   1.364 0.174173    
roe                       1.818e-02  4.957e-03   3.668 0.000312 ***
ros                      -7.828e-04  5.521e-04  -1.418 0.157763    
sales_mayor_promedio      4.502e-01  1.932e-01   2.330 0.020780 *  
sales_mayor_promedio_roe -4.848e-03  9.691e-03  -0.500 0.617402    
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.5153 on 203 degrees of freedom
Multiple R-squared:  0.1921,	Adjusted R-squared:  0.1722 
F-statistic: 9.653 on 5 and 203 DF,  p-v

El modelo extendido, que incluye las variables sales, roe, ros, sales_mayor_promedio y sales_mayor_promedio_roe, es globalmente significativo al 5%, indicando que el conjunto de estas variables explica de manera relevante la variación en el logaritmo del salario del CEO (lsalary). El R-squared de 0.1921 sugiere que aproximadamente el 19.21% de esta variabilidad es capturada por el modelo.

En cuanto a los regresores individuales, roe (retorno sobre el capital) mantiene su significancia estadística, mostrando que un mayor retorno sobre el capital se asocia positivamente con un salario más alto para el CEO. La nueva variable sales_mayor_promedio, que indica si las ventas de la empresa superan el promedio, también es estadísticamente significativa y tiene un efecto positivo considerable. Esto implica que los CEOs de empresas con ventas por encima del promedio tienden a recibir salarios sustancialmente más altos que aquellos en empresas con ventas promedio o por debajo, ceteris paribus.

Por otro lado, la variable sales (ventas) y ros (retorno sobre las acciones) no resultaron ser estadísticamente significativas en este modelo extendido, lo que sugiere que su influencia directa en el salario, una vez consideradas las otras variables y la dummy de ventas altas, es limitada. Finalmente, la variable de interacción sales_mayor_promedio_roe tampoco es significativa, lo que indica que el efecto del retorno sobre el capital en el salario del CEO no difiere de manera estadísticamente apreciable entre empresas con ventas altas y bajas.

3. (15 puntos) Realiza un test de hipotesis para evaluar si las dos variables que se agregaron son
significativas para el modelo simultaneamente. Use significancia del 5%.


In [23]:
anova_test <- anova(model_salary, model_salary_extended)
print("Resultado del test ANOVA para significancia conjunta:")
print(anova_test)

# Extraer el p-valor del test F
p_value_anova <- anova_test$`Pr(>F)`[2]

# Nivel de significancia
alfa <- 0.05

cat("\n--- Interpretación del Test F ---\n")
cat(paste("P-valor del test F para las variables añadidas: ", round(p_value_anova, 4), "\n"))

if (p_value_anova < alfa) {
  cat(paste("Dado que el p-valor (", round(p_value_anova, 4), ") es menor que el nivel de significancia (", alfa, "), rechazamos la hipótesis nula.", "\n"))
  cat("Esto significa que las variables 'sales_mayor_promedio' y 'sales_mayor_promedio_roe' son conjuntamente significativas para el modelo al 5% de significancia.", "\n")
} else {
  cat(paste("Dado que el p-valor (", round(p_value_anova, 4), ") es mayor o igual que el nivel de significancia (", alfa, "), no rechazamos la hipótesis nula.", "\n"))
  cat("Esto significa que las variables 'sales_mayor_promedio' y 'sales_mayor_promedio_roe' NO son conjuntamente significativas para el modelo al 5% de significancia.", "\n")
}

[1] "Resultado del test ANOVA para significancia conjunta:"
Analysis of Variance Table

Model 1: lsalary ~ sales + roe + ros
Model 2: lsalary ~ sales + roe + ros + sales_mayor_promedio + sales_mayor_promedio_roe
  Res.Df    RSS Df Sum of Sq      F   Pr(>F)   
1    205 57.401                                
2    203 53.905  2    3.4957 6.5821 0.001699 **
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

--- Interpretación del Test F ---
P-valor del test F para las variables añadidas:  0.0017 
Dado que el p-valor ( 0.0017 ) es menor que el nivel de significancia ( 0.05 ), rechazamos la hipótesis nula. 
Esto significa que las variables 'sales_mayor_promedio' y 'sales_mayor_promedio_roe' son conjuntamente significativas para el modelo al 5% de significancia. 


4. (15 puntos) Volviendo a la regresión del salario en función de sales, roe y ros. Considere
dos grupos, Grupo 1: Aquellos clientes que sales es mayor al promedio, y Grupo 2: Aquellos
clientes que sales es menor al promedio. Realice un test de hipotesis para grupos al 5% de
significancia. Use el metodo que más le acomode.

In [24]:
df_ceosal1$sales_D <- df_ceosal1$sales * df_ceosal1$sales_mayor_promedio
df_ceosal1$roe_D <- df_ceosal1$roe * df_ceosal1$sales_mayor_promedio
df_ceosal1$ros_D <- df_ceosal1$ros * df_ceosal1$sales_mayor_promedio

# Modelo no restringido (UR): Incluye la dummy 'sales_mayor_promedio' y sus interacciones
model_unrestricted <- lm(lsalary ~ sales + roe + ros + sales_mayor_promedio + sales_D + roe_D + ros_D, data = df_ceosal1)

# El modelo restringido (R) es el 'model_salary' que ya hemos definido:
# lsalary ~ sales + roe + ros
model_restricted <- model_salary

# Realizar el test F comparando ambos modelos
anova_groups <- anova(model_restricted, model_unrestricted)
print("Resultado del test ANOVA para la comparación de grupos (sales > promedio vs sales <= promedio):")
print(anova_groups)

# Extraer el p-valor del test F
p_value_anova_groups <- anova_groups$`Pr(>F)`[2]

# Nivel de significancia
alfa <- 0.05

cat("\n--- Interpretación del Test F para grupos ---\n")
cat(paste("P-valor del test F: ", round(p_value_anova_groups, 4), "\n"))

if (p_value_anova_groups < alfa) {
  cat(paste("Dado que el p-valor (", round(p_value_anova_groups, 4), ") es menor que el nivel de significancia (", alfa, "), rechazamos la hipótesis nula.", "\n"))
  cat("Esto significa que existe una diferencia significativa en la relación entre el salario del CEO y las variables explicativas entre los grupos con 'sales' mayores al promedio y 'sales' menores o iguales al promedio.\n")
  cat("Es decir, la inclusión de los términos de interacción y la dummy de sales por encima del promedio mejora significativamente el modelo, sugiriendo un efecto diferenciado para este grupo.\n")
} else {
  cat(paste("Dado que el p-valor (", round(p_value_anova_groups, 4), ") es mayor o igual que el nivel de significancia (", alfa, "), no rechazamos la hipótesis nula.", "\n"))
  cat("Esto significa que no hay evidencia suficiente para concluir que existe una diferencia significativa en la relación entre el salario del CEO y las variables explicativas entre los grupos con 'sales' mayores al promedio y 'sales' menores o iguales al promedio.\n")
}

[1] "Resultado del test ANOVA para la comparación de grupos (sales > promedio vs sales <= promedio):"
Analysis of Variance Table

Model 1: lsalary ~ sales + roe + ros
Model 2: lsalary ~ sales + roe + ros + sales_mayor_promedio + sales_D + 
    roe_D + ros_D
  Res.Df    RSS Df Sum of Sq      F    Pr(>F)    
1    205 57.401                                  
2    201 48.122  4     9.279 9.6893 3.474e-07 ***
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

--- Interpretación del Test F para grupos ---
P-valor del test F:  0 
Dado que el p-valor ( 0 ) es menor que el nivel de significancia ( 0.05 ), rechazamos la hipótesis nula. 
Esto significa que existe una diferencia significativa en la relación entre el salario del CEO y las variables explicativas entre los grupos con 'sales' mayores al promedio y 'sales' menores o iguales al promedio.
Es decir, la inclusión de los términos de interacción y la dummy de sales por encima del promedio mejora significativamente el modelo,

5. (15 puntos) Usando la regresión del salario en función de sales, roe y ros, se distinguen los
grupos mediante las dummies; indus, finance, consprod y utility, que corresponden al tipo de
empresa. Realice mediante test de Chow, con 5% de significancia, si los modelos para cada
grupo son iguales.

In [25]:
# Variables independientes en el modelo base (excluyendo el intercepto)
k <- 3

# Crear subconjuntos de datos para cada tipo de industria
df_indus <- subset(df_ceosal1, indus == 1)
df_finance <- subset(df_ceosal1, finance == 1)
df_consprod <- subset(df_ceosal1, consprod == 1)
df_utility <- subset(df_ceosal1, utility == 1)

# Lista para almacenar los SSR individuales y los modelos
ssr_individuales <- c()
modelos_individuales <- list()

# Número de grupos con suficientes observaciones
num_grupos_validos <- 0

# Procesar grupo 'indus'
if (nrow(df_indus) > k + 1) {
    model_indus <- lm(lsalary ~ sales + roe + ros, data = df_indus)
    ssr_individuales["indus"] <- sum(residuals(model_indus)^2)
    modelos_individuales[["indus"]] <- model_indus
    num_grupos_validos <- num_grupos_validos + 1
} else {
    print("Advertencia: El grupo 'indus' tiene muy pocas observaciones para una regresión válida.")
}

# Procesar grupo 'finance'
if (nrow(df_finance) > k + 1) {
    model_finance <- lm(lsalary ~ sales + roe + ros, data = df_finance)
    ssr_individuales["finance"] <- sum(residuals(model_finance)^2)
    modelos_individuales[["finance"]] <- model_finance
    num_grupos_validos <- num_grupos_validos + 1
} else {
    print("Advertencia: El grupo 'finance' tiene muy pocas observaciones para una regresión válida.")
}

# Procesar grupo 'consprod'
if (nrow(df_consprod) > k + 1) {
    model_consprod <- lm(lsalary ~ sales + roe + ros, data = df_consprod)
    ssr_individuales["consprod"] <- sum(residuals(model_consprod)^2)
    modelos_individuales[["consprod"]] <- model_consprod
    num_grupos_validos <- num_grupos_validos + 1
} else {
    print("Advertencia: El grupo 'consprod' tiene muy pocas observaciones para una regresión válida.")
}

# Procesar grupo 'utility'
if (nrow(df_utility) > k + 1) {
    model_utility <- lm(lsalary ~ sales + roe + ros, data = df_utility)
    ssr_individuales["utility"] <- sum(residuals(model_utility)^2)
    modelos_individuales[["utility"]] <- model_utility
    num_grupos_validos <- num_grupos_validos + 1
} else {
    print("Advertencia: El grupo 'utility' tiene muy pocas observaciones para una regresión válida.")
}

# Verificar si hay al menos dos grupos válidos para realizar el test de Chow
if (num_grupos_validos < 2) {
    stop("No hay suficientes grupos válidos para realizar el Test de Chow.")
}

# Suma de cuadrados de los residuos del modelo no restringido (suma de los SSR de cada grupo)
SSR_unrestricted <- sum(ssr_individuales)

# Suma de cuadrados de los residuos del modelo restringido (modelo original sin diferencias de grupo)
# model_salary ya está definido como lsalary ~ sales + roe + ros
SSR_restricted <- sum(residuals(model_salary)^2)

# Número total de observaciones
n <- nrow(df_ceosal1)

# Número de parámetros en cada modelo de grupo (k variables + 1 intercepto)
p <- k + 1

# Grados de libertad del numerador: (G - 1) * p
gl1 <- (num_grupos_validos - 1) * p

# Grados de libertad del denominador: n - G * p
gl2 <- n - (num_grupos_validos * p)

# Calcular el estadístico F del Chow test
F_statistic_chow <- ((SSR_restricted - SSR_unrestricted) / gl1) / (SSR_unrestricted / gl2)

# Calcular el p-valor
p_value_chow <- 1 - pf(F_statistic_chow, gl1, gl2)

# Nivel de significancia
alfa <- 0.05

cat("\n--- Resultados del Test de Chow ---\n")
cat(paste("SSR Restringido (modelo completo): ", round(SSR_restricted, 4), "\n"))
cat(paste("SSR No Restringido (suma de modelos por grupo): ", round(SSR_unrestricted, 4), "\n"))
cat(paste("Estadístico F del Chow test: ", round(F_statistic_chow, 4), "\n"))
cat(paste("Grados de libertad del numerador (gl1): ", gl1, "\n"))
cat(paste("Grados de libertad del denominador (gl2): ", gl2, "\n"))
cat(paste("P-valor del Test de Chow: ", round(p_value_chow, 4), "\n"))

if (p_value_chow < alfa) {
    cat(paste("Dado que el p-valor (", round(p_value_chow, 4), ") es menor que el nivel de significancia (", alfa, "), rechazamos la hipótesis nula.", "\n"))
    cat("Esto significa que los modelos de regresión son significativamente diferentes entre los grupos de la industria.\n")
} else {
    cat(paste("Dado que el p-valor (", round(p_value_chow, 4), ") es mayor o igual que el nivel de significancia (", alfa, "), no rechazamos la hipótesis nula.", "\n"))
    cat("Esto significa que no hay evidencia suficiente para concluir que los modelos de regresión son significativamente diferentes entre los grupos de la industria.\n")
}

# Opcional: Imprimir resúmenes de los modelos individuales si se desea revisar
# for (name in names(modelos_individuales)) {
#   print(paste("\n--- Resumen del Modelo para el grupo", name, "---"))
#   print(summary(modelos_individuales[[name]]))
# }


--- Resultados del Test de Chow ---
SSR Restringido (modelo completo):  57.4012 
SSR No Restringido (suma de modelos por grupo):  46.1153 
Estadístico F del Chow test:  3.9361 
Grados de libertad del numerador (gl1):  12 
Grados de libertad del denominador (gl2):  193 
P-valor del Test de Chow:  0 
Dado que el p-valor ( 0 ) es menor que el nivel de significancia ( 0.05 ), rechazamos la hipótesis nula. 
Esto significa que los modelos de regresión son significativamente diferentes entre los grupos de la industria.


El Test de Chow ha sido ejecutado satisfactoriamente. Los resultados muestran un estadístico F de 3.9361 con un p-valor de 0.

Dado que el p-valor 0 es significativamente menor que el nivel de significancia de 0.05, rechazamos la hipótesis nula.